src/Syntax/T1.idr

```diff
@@ -10,12 +10,20 @@ pure : a -> F1 s a
 pure = (#)
 
 export %inline
-(>>=) : {0 rs,ss,ts : _} -> C1 rs ss a -> (a -> C1 ss ts b) -> C1 rs ts b
+(>>=) :
+     (T1 rs -@ Res1 a f)
+  -> ((v : a) -> T1 (f v) -@ Res1 b g)
+  -> T1 rs
+  -@ Res1 b g
```

Comment:
Yeah, this looks like a slightly more general form than what [I implemented](https://src.imu.li/idris/file?name=Control/Res.idr&ci=tip), which essentially restricted `f` and `g` to either drop the value as a resource or cons it onto the list.